src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -947,6 +947,1128 @@ public static class Extensions
+                    IL_0013: newobj instance void [netstandard]System.Action::.ctor(object, native int)
+                    IL_0018: stloc.0
+                    IL_0019: ldstr "Foo()"
+                    IL_001e: call void [netstandard]System.Console::WriteLine(string)
+                    IL_0023: dup
+                    IL_0024: ldfld int32 IntExt/'<>c__DisplayClass1_0'::b
+                    IL_0029: ldc.i4.1
+                    IL_002a: add
+                    IL_002b: stloc.1
+                    IL_002c: ldloc.1
+                    IL_002d: stfld int32 IntExt/'<>c__DisplayClass1_0'::b
+                    IL_0032: ldloc.0
+                    IL_0033: ret
+                } // end of method IntExt::Foo
+            } // end of class IntExt
+            """
+        );
+
+        var tree = comp.SyntaxTrees[0];
+        var model = comp.GetSemanticModel(tree);
+        var extension1 = tree.GetRoot().DescendantNodes().OfType<ExtensionDeclarationSyntax>().First();
+        var symbol1 = model.GetDeclaredSymbol(extension1);
+        var sourceExtension1 = symbol1.GetSymbol<SourceNamedTypeSymbol>();
+        Assert.Equal("<>E__0", symbol1.MetadataName);
+        Assert.Equal("IntExt.<>E__0", symbol1.ToTestDisplayString());
+    }
+
+    [Fact]
+    [WorkItem("https://github.com/dotnet/roslyn/issues/78042")]
+    public void ExtensionWithCapturing3()
+    {
+        var src = """
+using System.Runtime.CompilerServices;
+using System.Collections.Generic;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+Console.WriteLine("Hello, World!");
+
+namespace System.Linq
+{
+    public static partial class AsyncEnumerable
+    {
+        extension<TSource>(IAsyncEnumerable<TSource> source)
+        {
+            /// <summary>
+            /// Casts the elements of an <see cref="IAsyncEnumerable{Object}"/> to the specified type.
+            /// </summary>
+            /// <typeparam name="TResult">The type to cast the elements of source to.</typeparam>
+            /// <returns>An <see cref="IAsyncEnumerable{TResult}"/> that contains each element of the source sequence cast to the <typeparamref name="TResult"/> type.</returns>
+            public IAsyncEnumerable<TResult> Cast<TResult>() // satisfies the C# query-expression pattern
+            {
+                return Impl(source, default);
+
+                static async IAsyncEnumerable<TResult> Impl(
+                    IAsyncEnumerable<TSource?> source,
+                    [EnumeratorCancellation] CancellationToken cancellationToken)
+                {
+                    await foreach (object? item in source.WithCancellation(cancellationToken))
+                    {
+                        yield return (TResult)item!;
+                    }
+                }
+            }
+        }
+    }
+}
+""";
+
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net90);
```

Comment:
>targetFramework: TargetFramework.Net90 [](http://example.com/codeflow?start=42&length=38)

I think using explicit target framework is the source of the CI failure. You should be able to reproduce it locally if you run the test while targeting net472 environment. Consider removing this parameter. Otherwise, if this target framework is required, try specifying ```verify: Verification.FailsPEVerify``` as an argument for ```CompileAndVerify``` call below.